File: hub/src/light/effect/compiled_effect.py
from light.effect.resolved_effect import ResolvedEffect
from light.effect.effect_priority import EffectPriority

class CompiledEffect(ResolvedEffect):
    def __init__(self, hitTypes, effect, dt, duration, priority=EffectPriority.ANY, isAmbient=False, isModifier=False, noteTime=0, t=0, ambientDuration=0):
        super().__init__(effect.rgbw, effect.amplitude, effect.ledPositions)
        self.dt = dt
        self.duration = duration
        self.hitTypes = hitTypes
        self.priority = priority
        self.isAmbient = isAmbient
        self.isModifier = isModifier
        self.noteTime = noteTime
        self.t = t
        self.ambientDuration = ambientDuration
    
    def clone(self):
        return CompiledEffect(
            self.hitTypes, 
            ResolvedEffect(
                self.rgbw,
                self.amplitude,
                self.ledPositions), 
            self.dt,
            self.duration,
            self.priority,
            self.isAmbient,
            self.isModifier,
            self.noteTime,
            self.t,
            self.ambientDuration)

    def toString(self):
        return super().toString()

File: hub/drumlite-js/src/effect/CompiledEffect.ts
import { HitType } from "../midi/HitType"
import ResolvedEffect from "./ResolvedEffect"
import { EffectPriority } from "./EffectPriority"

export default class CompiledEffect extends ResolvedEffect {
    dt: number;
    duration: number;
    hitTypes: HitType[];
    priority: EffectPriority;
    isAmbient: boolean;
    isModifier: boolean;
    noteTime: number;
    t: number;
    ambientDuration: number;

    constructor(hitTypes: HitType[], effect: ResolvedEffect, dt: number, duration: number, priority=EffectPriority.ANY, isAmbient=false, isModifier=false, noteTime=0, t=0, ambientDuration=0) {
        super(effect.rgb, effect.amplitude, effect.ledPositions);
        this.dt = dt
        this.duration = duration
        this.hitTypes = hitTypes
        this.priority = priority
        this.isAmbient = isAmbient
        this.isModifier = isModifier
        this.noteTime = noteTime
        this.t = t
        this.ambientDuration = ambientDuration
    }
}
